frontend/models/product-list/component/product-list-ancestor.ts

```diff
@@ -0,0 +1,85 @@
+import { getProductListTitle } from '@helpers/product-list-helpers';
+import type { DeviceWiki } from '@lib/ifixit-api/devices';
+import type { ProductListFieldsFragment } from '@lib/strapi-sdk';
+import { z } from 'zod';
+import { ProductListType, ProductListTypeSchema } from './product-list-type';
+import { productListTypeFromStrapi } from './product-list-type.server';
+
+export type ProductListAncestor = z.infer<typeof ProductListAncestorSchema>;
+
+export const ProductListAncestorSchema = z.object({
+   deviceTitle: z.string().nullable(),
+   title: z.string(),
+   type: ProductListTypeSchema,
+   handle: z.string(),
+});
+
+export function createProductListAncestorsFromStrapiOrDeviceWiki(
+   strapiProductList: ProductListFieldsFragment | null | undefined,
+   deviceWiki: DeviceWiki | null
+): ProductListAncestor[] {
+   if (strapiProductList?.parent) {
+      return createProductListAncestorsFromStrapi(strapiProductList?.parent);
+   }
+   if (deviceWiki?.ancestors) {
+      return createProductListAncestorsFromDeviceWiki(deviceWiki?.ancestors);
+   }
+   return [];
+}
+
+function createProductListAncestorsFromStrapi(
+   parent: ProductListFieldsFragment['parent']
+): ProductListAncestor[] {
+   const attributes = parent?.data?.attributes;
+   if (attributes == null) {
+      return [];
+   }
+   const ancestors = createProductListAncestorsFromStrapi(attributes.parent);
+
+   const type = productListTypeFromStrapi(attributes.type);
+
+   return ancestors.concat({
+      deviceTitle: attributes.deviceTitle ?? null,
+      title: getProductListTitle({
+         title: attributes.title,
+         type,
+      }),
+      type: productListTypeFromStrapi(attributes.type),
```

Comment:
It seems like we already compute this above on line 39. I see that it was this way in the previous implementation, but I don't see harm in reusing the variable.

```suggestion
      type,
```